### PR TITLE
Build new website for documentation

### PR DESCRIPTION
**Pull request by JanMa**
This PR adds a new website built with [Docusaurus](https://docusaurus.io) in the `/website` folder.

I have tried to integrate our Markdown docs with the already existing [website WIP](https://github.com/openbao/website) and was not able to produce anything satisfying. So I had a look at how OpenTofu built their website and it turns out they are using Docusaurus instead of NextJS. 

Building upon an example Docusaurus  I was able to produce some great results and made a lot of progress in rewriting the documentation. You can see this PR deployed to GitHub Pages from my fork here:
https://janma.github.io/openbao

I decided to leave the default landing page of the example site untouched for this PR, so  the main thing that's left to do is to create a proper landing page.

### CODE REVIEW

File: website/docusaurus.config.ts [hunk 1]
```diff
@@ -0,0 +1,157 @@
+import { themes as prismThemes } from "prism-react-renderer";
+import type { Config } from "@docusaurus/types";
+import type * as Preset from "@docusaurus/preset-classic";
+import { includeMarkdown } from "@hashicorp/remark-plugins";
+import path from "path";
+
+const config: Config = {
+  title: "OpenBao",
+  tagline: "Manage, Store and distribute sensitive data",
```
Comment (cipherboy) [inline website/docusaurus.config.ts:9]:
```suggestion
  tagline: "Manage, store, and distribute sensitive data",
```

**Comment by joewxboy** [comment]
This is looking good.  I should open another issue for automation that scans the built website looking for broken hyperlinks and orphaned pages.  We do something similar with the Open Horizon docs.  If you know of similar tools, please let me know.

File: website/Makefile [hunk 2]
```diff
@@ -1,58 +0,0 @@
-######################################################
```
[the remark this answers is not in the document]
Comment (JanMa) [reply website/Makefile:1]: You can just run `npm run start` to start a local dev instance. I'm not sure if we really need a Makefile or docker container for this?  I also documented this in the website README

[the remark this answers is not in the document]
Comment (JanMa) [reply website/Makefile:1]: On second thought, I might as well just add one. It's not a lot of code

**Comment by JanMa** [comment]
> This is looking good.  I should open another issue for automation that scans the built website looking for broken hyperlinks and orphaned pages.  We do something similar with the Open Horizon docs.  If you know of similar tools, please let me know.

During the build process Docusaurus warns you about broken links and anchors. You can also make the build fail if it finds any.

I intentionally disabled this for now since we know there are still some broken links, but we can definitely add some checks in the future once we cleaned that up.

**Comment by cipherboy** [comment]
Hmm, I'm having a little bit of a problem starting the Node server as of 90b3bc6115853f46ec4147ce8304f81e9e5882ac:

```
[cipherboy@lp10 website]$ make
npm WARN deprecated stable@0.1.8: Modern JS already guarantees Array#sort() is a stable sort, so this library is deprecated. See the compatibility table on MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/sort#browser_compatibility
npm WARN deprecated mkdirp@0.3.0: Legacy versions of mkdirp are no longer supported. Please update to mkdirp 1.x. (Note that the API surface has changed to use Promises in 1.x.)

added 1286 packages, and audited 1287 packages in 6s

324 packages are looking for funding
  run `npm fund` for details

found 0 vulnerabilities

> website@0.0.0 start
> docusaurus start

[INFO] Starting the development server...

Error: Docusaurus could not load module at path "/home/cipherboy/GitHub/openbao/openbao/website/docusaurus.config.ts"
Cause: Cannot find module '@hashicorp/remark-plugins'
Require stack:
- /home/cipherboy/GitHub/openbao/openbao/website/docusaurus.config.ts
    at loadFreshModule (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/utils/lib/moduleUtils.js:36:15)
    at loadSiteConfig (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/core/lib/server/config.js:35:62)
    at async loadContext (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/core/lib/server/index.js:31:63)
    at async load (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/core/lib/server/index.js:74:21)
    at async Command.start (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/core/lib/commands/start.js:44:19) {
  [cause]: Error: Cannot find module '@hashicorp/remark-plugins'
  Require stack:
  - /home/cipherboy/GitHub/openbao/openbao/website/docusaurus.config.ts
      at Module._resolveFilename (node:internal/modules/cjs/loader:1144:15)
      at Function.resolve (node:internal/modules/helpers:187:19)
      at _resolve (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/jiti/dist/jiti.js:1:251148)
      at jiti (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/jiti/dist/jiti.js:1:253746)
      at /home/cipherboy/GitHub/openbao/openbao/website/docusaurus.config.ts:4:22
      at evalModule (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/jiti/dist/jiti.js:1:256443)
      at jiti (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/jiti/dist/jiti.js:1:254371)
      at loadFreshModule (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/utils/lib/moduleUtils.js:33:16)
      at loadSiteConfig (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/core/lib/server/config.js:35:62)
      at async loadContext (/home/cipherboy/GitHub/openbao/openbao/website/node_modules/@docusaurus/core/lib/server/index.js:31:63) {
    code: 'MODULE_NOT_FOUND',
    requireStack: [
      '/home/cipherboy/GitHub/openbao/openbao/website/docusaurus.config.ts'
    ]
  }
}

[INFO] Docusaurus version: 3.1.1
Node version: v20.10.0
make: *** [Makefile:6: start] Error 1
```

Does it need to be included in `package.json` perhaps?

**Comment by JanMa** [comment]
> Hmm, I'm having a little bit of a problem starting the Node server as of [90b3bc6](https://github.com/openbao/openbao/commit/90b3bc6115853f46ec4147ce8304f81e9e5882ac):

Weird. I tried a fresh install locally by removing all node modules and it worked. I'll have a look later if I missed something

**Review by cipherboy — APPROVED** [review]
Ah, much better! Thank you Jan, this looks great!

**Review by naphelps — APPROVED** [review]

**Pull request merged**